[SMTChecker] Fixing type conversion from StringLiteral to FixedBytes (#10490)
ethereum/solidity: Changelog.md

 * SMTChecker: Fix internal compiler error when doing bitwise compound assignment with string literals.
 * SMTChecker: Fix internal error when trying to generate counterexamples with old z3.
 * SMTChecker: Fix segmentation fault that could occur on certain SMT-enabled sources when no SMT solver was available.

>  * SMTChecker: Fix incorrect analysis of string literals cast to byte arrays.

Reviewer: ```suggestion
 * SMTChecker: Fix cast from string literals to byte arrays.
```

Author: Fixed.
